src/lib/storage/index/adaptive_radix_tree/adaptive_radix_tree_nodes.cpp

```diff
@@ -141,8 +141,8 @@ AbstractIndex::Iterator ARTNode16::_delegate_to_child(
     const std::function<Iterator(std::iterator_traits<std::array<uint8_t, 16>::iterator>::difference_type, size_t)>&
         function) const {
   auto partial_key = key[depth];
-  auto partial_key_iterator = std::lower_bound(_partial_keys.begin(), _partial_keys.end(), partial_key);
-  auto partial_key_pos = std::distance(_partial_keys.begin(), partial_key_iterator);
+  const auto* const partial_key_iterator = std::lower_bound(_partial_keys.begin(), _partial_keys.end(), partial_key);
```

Comment:
Iterator as a pointer?

Reply:
An iterator does not have to be class or a struct.

> Iterators are a generalization of pointers that allow a C++ program to work with different data structures (for example, containers and ranges) in a uniform manner.

As above, I am not sure what the intention of this is, but I don't see real harm being done.